[wasm] Don't use workload for tfm < net6.0

radical opened the pull request:
If the project requires the workload (eg. for AOT), but the project targets <net6.0, then error out before `Publish`.

To support this, the Wasm.Build.Tests not support testing with sdk, with no workload installed.

Currently, Wasm.Build.Tests are run in two modes:

1. sdk(`$(SdkVersionForWorkloadTesting)`)+workload installed
2. sdk(from global.json)+EMSDK+LocalBuild targets

this commit changes (2) to:

2. sdk(`$(SdkVersionForWorkloadTesting)`), no workload installed, +
   EMSDK+LocalBuild targets

This makes it closer to what a user would actually use.

Also, for the workload tests, this removes `$(WasmNativeWorkload)`
always being set to `true`. The workload should be setting that, and not
the tests.

Fixes https://github.com/dotnet/runtime/issues/54308

akoeplinger commented on src/mono/nuget/Microsoft.NET.Workload.Mono.Toolchain.Manifest/WorkloadManifest.targets.in:
```diff
@@ -1,11 +1,16 @@
         <RuntimePackInWorkloadVersion>${PackageVersion}</RuntimePackInWorkloadVersion>
+
+        <!-- not supported on < 6.0 -->
+        <UsingBrowserRuntimeWorkload Condition="'$(RuntimeIdentifier)' == 'browser-wasm' and !$([MSBuild]::VersionGreaterThanOrEquals('$(TargetFrameworkVersion)', '6.0'))">false</UsingBrowserRuntimeWorkload>
```
'$(TargetFrameworkIdentifier)' == '.NETCoreApp' and
```suggestion
        <UsingBrowserRuntimeWorkload Condition="'$(RuntimeIdentifier)' == 'browser-wasm' and '$(TargetFrameworkIdentifier)' == '.NETCoreApp' and  !$([MSBuild]::VersionGreaterThanOrEquals('$(TargetFrameworkVersion)', '6.0'))">false</UsingBrowserRuntimeWorkload>
```

akoeplinger commented on src/tests/BuildWasmApps/testassets/Blazor_net50/wwwroot/css/open-iconic/FONT-LICENSE:
```diff
@@ -0,0 +1,86 @@
+SIL OPEN FONT LICENSE Version 1.1
```
I'd really like to avoid checking in bootstrap/fonts/css etc. assets into the repo. Can we make a minimal test instead?

radical replied:
This is the blazor template app, but I guess since we aren't running the app, we could drop these.

lewing reviewed (commented):
I think we probably want an error/warning that explains this?

radical commented:
> I think we probably want an error/warning that explains this?

I've made it an error when the project requires a workload, but we have to disable it due fo tom<net6.0 . And emitting the error before `publish`, since that's when the workload would have kicked in. It will be changed to run before `build`, once we move to that from `publish`.

akoeplinger commented on src/tests/BuildWasmApps/testassets/Blazor_net50/Program.cs:
```diff
@@ -0,0 +1,25 @@
+        public static async Task Main(string[] args)
+        {
+            var builder = WebAssemblyHostBuilder.CreateDefault(args);
+            builder.RootComponents.Add<App>("#app");
```
nit: there are still a bunch of .css, .razor, weather.json, favicon.ico etc assets that don't seem relevant if we're not running the app so we could get rid of them too as having this `Main()` here should be enough to test the workload behavior right?

radical replied:
Removing the razor files, it ends up with pretty much an empty Main. I think, it would be preferable to have the razor files around at least, so they can get processed by the build. I'll remove the other files that are needed for running.

radical commented:
I think this needs some fixes right now, so marked it no-merge.

radical commented:
I had to make bigger changes to test the case of testing with latest sdk, and no workload installed. And that exposed other issues, so there are new changes here now. And I'll re-request for review, once I have it working on helix too.

radical commented:
/azp run runtime,runtime-staging

radical committed:
Bump sdk used for testing to 6.0.100-rc.1.21402.6

radical committed:
InstallWorkloadFromArtifacts: Add OnlyUpdateManifests parameter

radical committed:
[wasm] Run Wasm.Build.Tests with sdk/no-workload/+EMSDK

Currently, Wasm.Build.Tests are run in two modes:

1. sdk(`$(SdkVersionForWorkloadTesting)`)+workload installed
2. sdk(from global.json)+EMSDK+LocalBuild targets

this commit changes (2) to:

2. sdk(`$(SdkVersionForWorkloadTesting)`), no workload installed, +
   EMSDK+LocalBuild targets

This makes it closer to what a user would actually use.

Also, for the workload tests, this removes `$(WasmNativeWorkload)`
always being set to `true`. The workload should be setting that, and not
the tests.

radical committed:
Wasm.Build.Tests: Add test for net5.0 blazor projects

radical committed:
Disable unrelated builds for now

radical commented:
Remaining test failure is waiting on https://github.com/dotnet/runtime/pull/56606 .

radical committed:
Revert "Disable unrelated builds for now"

This reverts commit 6fa0efe2ffcfdbcfaba3bdf519a7717809c22672.

radical committed:
Merge remote-tracking branch 'origin/main' into wasm-skip-workload-5.0

radical committed:
Rename OnlyWasmBuildTests -> TestWasmBuildTests

radical committed:
Merge remote-tracking branch 'origin/main' into wasm-skip-workload-5.0

radical commented:
Blocked on https://github.com/dotnet/runtime/pull/56974 to get the emsdk update.

radical committed:
Merge remote-tracking branch 'origin/main' into wasm-skip-workload-5.0

radical committed:
Merge remote-tracking branch 'origin/main' into wasm-skip-workload-5.0

radical committed:
Merge remote-tracking branch 'origin/main' into wasm-skip-workload-5.0

radical commented:
Installer test failure is https://github.com/dotnet/runtime/issues/57243 , unrelated to this PR.

radical commented:
This could use some testing with VS, runtime/library tests, and blazorwasm projects.

lewing reviewed (commented):
Looking good, I'll do some manual testing then approve.
